server/src/action/openweather/openweather.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@nestjs/common';
+import { ActionService } from '../action.service';
+import { EventEmitter2, OnEvent } from '@nestjs/event-emitter';
+import { OpenweatherDto } from './dto/openweather.dto';
+import axios from 'axios';
+
+@Injectable()
+export class OpenweatherService {
+    constructor(
+        private actionService: ActionService,
+        private eventEmitter: EventEmitter2,
+    ) {
+        this.eventEmitter.on("Weather.struct", (struct: OpenweatherDto) => {
+            struct.city = "Nantes";
+            struct.operation = true;
```

Comment:
what is operation ?

Reply:
operation is a boolean, for me it could be cool for the user to say if it's superior or inferior. So for example if the user want to trigger the action when it's superior to 20° then it will choose true otherwise false for inferior.

Reply:
maybe change the name to superior, as it will be a little beat better for the user to understand

Reply:
done